client/src/lib.rs

```diff
@@ -86,4 +87,63 @@ impl ClientContract {
         }
         result
     }
-}
\ No newline at end of file
+
+    #[allow(dead_code)]
+    pub fn get_received_val(&self, nonce: U128) -> String {
+        let nonce_u128: u128 = nonce.into();
+        self.received.get(&nonce_u128).unwrap_or("-1".to_string())
+    }
+
+    #[allow(dead_code)]
+    pub fn reset(&mut self) {
```

Comment:
What is the purpose of this reset function? The oracle contract will start to reject the requests if the nonce is reset for the account. The purpose of the nonce is to be used only once, it's should not be reset.